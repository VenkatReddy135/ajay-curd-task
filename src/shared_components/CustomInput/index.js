/* eslint-disable react/destructuring-assignment */
import React from "react";
import PropTypes from "prop-types";

/**
 *
 * @param {object} props all props
 *@function CustomInput
 * @returns {React.createElement} CustomToaster
 */
export default function CustomInput(props) {
  return (
    <input
      type={props.inputType}
      placeholder={props.placeHolderText}
      onChange={props.handleOnChange}
      className={props.inputClass}
      value={props.inputValue}
    />
  );
}

CustomInput.propTypes = {
  inputType: PropTypes.string.isRequired,
  handleOnChange: PropTypes.func,
  placeHolderText: PropTypes.string,
  inputClass: PropTypes.string,
  inputValue: PropTypes.any,
};

CustomInput.defaultProps = {
  inputType: "",
  placeHolderText: "",
};
