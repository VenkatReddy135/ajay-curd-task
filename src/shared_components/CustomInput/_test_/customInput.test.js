import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import CustomInput from "../index";
import { cleanup } from "@testing-library/react";
import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-17-updated";
Enzyme.configure({ adapter: new Adapter() });

let wrapper;

beforeEach(() => {
  wrapper = shallow(<CustomInput />);
});
const props = {
  inputType: "",
  placeHolderText: "",
};
describe("Custom Input Component Testing", () => {
  it("renders component", () => {
    const tree = renderer.create(<CustomInput {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<CustomInput {...props} />, div);
  });
  it("should have input element", () => {
    console.log(wrapper);
    expect(wrapper.find("input")).toHaveLength(1);
  });

  afterEach(() => {
    cleanup();
  });
});
