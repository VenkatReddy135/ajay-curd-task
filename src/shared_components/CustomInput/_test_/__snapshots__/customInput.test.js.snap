// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Input Component Testing renders component 1`] = `
<input
  placeholder=""
  type=""
/>
`;
